feat: add config for bytecode comparison for the EasyTrack's contracts

Alexvozhak opened the pull request:
- Bump Hardhat to latest v2.24 (Prague hardfork support added in [v2.23)](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.23.0)
![image](https://github.com/user-attachments/assets/cfdd3c4e-6602-45e4-a1dd-8c9ca538458b)

- Add a warning and auto-skip of source verification when a flattened contract (e.g. EasyTrack) is detected
![image](https://github.com/user-attachments/assets/be61183e-187c-400d-a854-d3dd404025b3)


The entire backstory is [here](https://github.com/lidofinance/diffyscan/issues/6#issuecomment-2867600361)

tamtamchik commented on diffyscan/diffyscan.py:
```diff
@@ -272,14 +272,15 @@ def process_config(
-                    recursive_parsing,
-                    unify_formatting,
-                )
+                # Temportary disable source code diff
```
Maybe make it optional via run params?

TheDZhon replied:
huh?

tamtamchik reviewed (approved):
Maybe hide `run_source_diff` under the run parameter, otherwise looks good! 🎖️

TheDZhon reviewed (changes requested):
:eyes:

tamtamchik commented on diffyscan/diffyscan.py:
```diff
@@ -386,5 +391,14 @@ def main():
+        return False
+
+    filename, _ = files[0]
+    return ".sol" not in filename and "/" not in filename
```
It would be nice to add an example above, because the code is not for IQ50 vibe coders... how do we distinguish the single contract file from the monolithic? Does it have a special name or pattern?

Alexvozhak replied:
[done](https://github.com/Alexvozhak/diffyscan/pull/3/commits/a16e46698d6b4bc5f0a698f253557419391bc6d0)

tamtamchik reviewed (approved):
Approved, but it would be helpful to have some clarification on how we distinguish monolithic files from single-file contracts to avoid lurking in the shadows and googling or talking with AI.

Alexvozhak committed:
fix: remove Etherscan API key from logs

TheDZhon committed:
Merge pull request #85 from Alexvozhak/security/etherscan-key-disclosure

fix: remove Etherscan API key from logs

TheDZhon committed:
Merge pull request #78 from Alexvozhak/hoodi

fix: the hoodi configs added

TheDZhon committed:
fix: bump hardhat version to support pectra

TheDZhon commented:
Plz retarget the PR on the main repo

tamtamchik committed:
Merge pull request #86 from lidofinance/chore/bump-hardhat

Alexvozhak committed:
Merge branch 'main' into hoodi

Alexvozhak committed:
Merge branch 'hoodi' into feat/easy_track

Alexvozhak committed:
fix: remove accidental slice in solcInput key

Alexvozhak committed:
fix: improve integer encoding to handle string inputs

Alexvozhak committed:
feat: added EasyTrack's contracts handling

Alexvozhak commented:
> Plz retarget the PR on the main repo

[done](https://github.com/lidofinance/diffyscan/pull/87)

Alexvozhak committed:
feat: add source code comparison toggle

This allows to disable source code comparison via config when needed
(e.g. for monolithic contracts like EasyTrack)

Alexvozhak committed:
fix: enhance calldata handling by checking for constructor in ABI
Now missing calldata in config is not an error if constructor has no arguments in ABI

Alexvozhak committed:
fix: the warning message which occurs when source comparison is disabled

Co-authored-by: Eugene Mamin <TheDZhon@gmail.com>

Alexvozhak committed:
fix: renamed the config flag fail_on_comparison_error to fail_on_bytecode_comparison_error
